cli/command/registry.go

```diff
@@ -106,7 +106,12 @@ func ConfigureAuth(cli Cli, flUser, flPassword string, authconfig *registrytypes
 	if flUser = strings.TrimSpace(flUser); flUser == "" {
 		if isDefaultRegistry {
 			// if this is a default registry (docker hub), then display the following message.
-			fmt.Fprintln(cli.Out(), "Login with your Docker ID to push and pull images from Docker Hub. If you don't have a Docker ID, head over to https://hub.docker.com to create one.")
+			fmt.Fprintln(cli.Out(), "Log in with your Docker ID or email address to push and pull images from Docker Hub. If you don't have a Docker ID, head over to https://hub.docker.com/ to create one.")
+			if os.Getenv("DOCKER_CLI_HINTS") != "false" {
```

Comment:
Maybe we should also handle `"no"`, like yaml does? I'm joking, this check comes from [compose-cli](https://github.com/docker/compose-cli/pull/2261/files#diff-a31a3a51fbc5e873ef63ca842afed4cf220a9ae7589559a856a503d6957d68dfL41).

I kinda like only one possible value here

Reply:
Another nice possibility is to flip the check, have a `DOCKER_NO_CLI_HINTS` and only check that the variable is defined

Reply:
Personally I'd prefer `DOCKER_NO_CLI_HINTS`.

Reply:
`DOCKER_DISABLE_CLI_HINTS` would be an option.

I think we came with `DOCKER_CLI_HINTS` in a similar fashion as `DOCKER_BUILDKIT`; initially that was "opt-in", but the intent was to make that "opt-out" after incubation.

`DOCKER_CLI_HINTS` (and `DOCKER_BUILDKIT`) is more agnostic w.r.t. it being enable/disable. And can start with "opt-in" (`DOCKER_CLI_HINTS=1`) and when it's the default, "opt-out" (`DOCKER_CLI_HINTS=0`), or for situations where the config would disable it and you wanted to enable it for a specific call (through env var), as `DOCKER_DISABLE_CLI_HINTS=false` is kinda awkward because of the double negative.

w.r.t. accepted values, if we _do_ look at value (beyond "non-empty"), I agree that we should use `ParseBool` if the value is non-empty, which would match other env-vars, such as `DOCKER_BUILDKIT`; https://github.com/docker/cli/blob/06de1f8d2c56737ea06a547197a5ae3701aa5397/cli/command/cli.go#L164-L171

Reply:
+1 to ParseBool and `DOCKER_NO_CLI_HINTS`

Reply:
We already have `DOCKER_CLI_HINTS` (see https://github.com/docker/for-mac/issues/6904 for evidence of people building on it); we might have to continue with that out of inertia, or at least propagate any changes we make here back to the Scout hints implementation (cc @eunomie)